opm/parser/eclipse/EclipseState/Schedule/Segment.hpp

```diff
@@ -0,0 +1,116 @@
+/*
+  Copyright 2015 SINTEF ICT, Applied Mathematics.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef SEGMENT_HPP_HEADER_INCLUDED
+#define SEGMENT_HPP_HEADER_INCLUDED
+
+#include <memory>
+
+#include <opm/parser/eclipse/Deck/DeckKeyword.hpp>
+#include <opm/parser/eclipse/Deck/DeckRecord.hpp>
+
+
+namespace Opm {
+
+    class Segment {
+    public:
+        Segment();
+
+        Segment(int segment_number, int branch, int outlet_segment, double length, double depth,
+                double internal_diameter, double roughness, double cross_area,
+                double volume, double length_x, double length_y, bool data_ready);
+
+        int& segmentNumber();
```

Comment:
> How do you think?

Don't know yet; but as you say it is currently a mix. Got to go now - continue tomorrow.